### PR TITLE
Remove CLI::Input and other refactorings

### PR DESCRIPTION
**Pull request by mvz**
#901.

### CODE REVIEW

File: lib/reek/cli/application.rb [hunk 1]
```diff
@@ -13,18 +13,31 @@ module CLI
...
+                                     sources: sources,
+                                     configuration: configuration)
       end
 
       def execute
-        command.execute self
+        command.execute
+      end
+
+      def sources
```
Comment (mvz) [inline lib/reek/cli/application.rb]: Maybe I should make this private too.

Comment (troessner) [inline lib/reek/cli/application.rb]: 👍

**Comment by troessner** [comment]
Very nice, love it!
From a PR point I think this is ready to merge after squashing.
Seems like we have mutant problem though. If you check out e.g. the build for [2.3](https://travis-ci.org/troessner/reek/jobs/122722995) you'll see this:

>> (382/552)  69% - killtime: 149.78s runtime: 6901.11s overhead: 6751.33s
(382/552)  69% - killtime: 149.78s runtime: 6902.11s overhead: 6752.33s
Your test run exceeded 120 minutes. 
One possible solution is to split up your test run.

I'm fine with merging it as it is but we then should look into the mutant problem.

**Comment by mvz** [comment]
> we then should look into the mutant problem.

Yes. We need to take another look at how mutant works and what it expects from our tests.

**Comment by mvz** [comment]
@troessner squashed.

**Pull request merged**